ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -16,13 +16,16 @@
 from mlagents.envs.base_unity_environment import BaseUnityEnvironment
 from mlagents.envs.exception import UnityEnvironmentException
 from mlagents.trainers import Trainer
+from mlagents.envs.sampler_class import SamplerManager
 from mlagents.trainers.ppo.trainer import PPOTrainer
 from mlagents.trainers.bc.offline_trainer import OfflineBCTrainer
 from mlagents.trainers.bc.online_trainer import OnlineBCTrainer
 from mlagents.trainers.meta_curriculum import MetaCurriculum
 
 
+
 class TrainerController(object):
+    # Type of reset_param_dict unspecified as typing library does not support heterogeneous dictionary types
```

Comment:
You should be able to use `Dict[str, Any]` for heterogenous values (or Dict[str, Union[type1, type1]] if you know the possible types). But I think the proto specifies string keys and float values: https://github.com/Unity-Technologies/ml-agents/blob/c025086818949707dc5cf47a47b10b86e3d42937/protobuf-definitions/proto/mlagents/envs/communicator_objects/environment_parameters_proto.proto#L9

Reply:
Trailing comment from old code, no longer relevant here, but type information is good to know.